### ISSUE TITLE
Country flags!

### ISSUE THREAD

**Issue by FANMixco** [microsoft/fluentui-emoji#40]
It's time that Microsoft brings county flags support. Pretty much other OSs like Android, iOS, macOS, etc. have emojis for each country, but for any mysterious reason Windows don't! 🇵🇱🇪🇸🇪🇺

**Comment by jgarcianewemage** [comment]
only flags available are the political kind approved by overlords

**Comment by yell0wsuit** [comment]
Quote from somewhere:

> The reason is a concession from Microsoft to the Chinese government. The CCP does not accept the use of Taiwan flag emoji. Microsoft decided not to display any emoji country flags at all and thus evaded a political position on the One China issue.

**Comment by FANMixco** [comment]
> Quote from somewhere:
> 
> > The reason is a concession from Microsoft to the Chinese government. The CCP does not accept the use of Taiwan flag emoji. Microsoft decided not to display any emoji country flags at all and thus evaded a political position on the One China issue.

I'm pro-Taiwan, but until this is solved, you can just restrict its creation or don't show it in other countries. What's more, if you feel this is discriminative, you have at least 10 other countries with limited recognition:

https://en.wikipedia.org/wiki/List_of_states_with_limited_recognition

**Comment by yell0wsuit** [comment]
I only quote the reason why Microsoft decides not to implement the flag emoji, so I don't understand why you gave my comment a dislike. Like, if Microsoft decides not to do that then you can't do anything about it. It's up to Microsoft to do that, not me.

**Comment by hkienle** [comment]
This official explanation is not particularly enlightening, but here it goes:

> Clippy isn’t included in the open source emoji set 😩. I know, and I’d personally love to but due to (legal speak, legal speak, legal speak), we can’t open-source trademarks. For the same reason, our **country flags**, video game, and technologist emoji also won’t be available.
https://medium.com/microsoft-design/designing-in-the-open-source-5c62be73a599

**Comment by FANMixco** [comment]
> This official explanation is not particularly enlightening, but here it goes:
> 
> > Clippy isn’t included in the open source emoji set 😩. I know, and I’d personally love to but due to (legal speak, legal speak, legal speak), we can’t open-source trademarks. For the same reason, our **country flags**, video game, and technologist emoji also won’t be available.
> > https://medium.com/microsoft-design/designing-in-the-open-source-5c62be73a599

The "official" explanation makes no sense: 
> ... we **can’t open-source trademarks**. For the same reason, our **country flags** ...

For a simple reason, the country flags don't exist in Windows!

<img width="275" alt="image" src="https://user-images.githubusercontent.com/1666156/192139000-ab0af160-eb64-4dfd-9cc7-7012599de2d9.png">

<img width="272" alt="image" src="https://user-images.githubusercontent.com/1666156/192139012-251fee01-7b39-47f1-959f-05dc57d9becb.png">

<img width="273" alt="image" src="https://user-images.githubusercontent.com/1666156/192139019-2c600598-d274-4533-9a2f-5d0d2571fd06.png">

<img width="444" alt="image" src="https://user-images.githubusercontent.com/1666156/192139034-4a22cb52-63f8-4dc2-99f6-a980f843f81a.png">

![Web capture_25-9-2022_122934_emojipedia org](https://user-images.githubusercontent.com/1666156/192139069-08e54701-04a5-40f9-a80c-2d7f083dc035.jpeg)

Either they are being planned or the "official" answer was wrongly written.

**Comment by wilsontulus** [comment]
Great idea! It's have been a long struggle to get flag emojis work natively on Windows, especially since Chromium browsers are known to only load system fonts through a whitelist (specified in the source code: https://github.com/chromium/chromium/blob/master/third_party/blink/renderer/platform/fonts/win/font_fallback_win.cc#L99) so simply by adding a third-party emoji font with flags wouldn't work. Only ways are by using Firefox (since the browser came with Twemoji which have every single flags in the emoji set) or by tweaking a Noto Color Emoji font to disguise as Segoe UI Emoji with the caveat of some applications (including Firefox, sadly 😓) not supporting it due to Google's decision of color font formats.
I hope this suggestion goes well in the near future 😊

**Comment by FANMixco** [comment]
> Great idea! It's have been a long struggle to get flag emojis work natively on Windows, especially since Chromium browsers are known to only load system fonts through a whitelist (specified in the source code: https://github.com/chromium/chromium/blob/master/third_party/blink/renderer/platform/fonts/win/font_fallback_win.cc#L99) so simply by adding a third-party emoji font with flags wouldn't work. Only ways are by using Firefox (since the browser came with Twemoji which have every single flags in the emoji set) or by tweaking a Noto Color Emoji font to disguise as Segoe UI Emoji with the caveat of some applications (including Firefox, sadly 😓) not supporting it due to Google's decision of color font formats. I hope this suggestion goes well in the near future 😊

Please vote for the idea :D

**Comment by Mgrmjp** [comment]
Maybe in Windows 12!

**Comment by TaylorTWBrown** [comment]
Either give us flag emojis, or I will personally protest on the lawn of Bill Gates, Steve Ballmer, and Satya Nadella. This is nuts! At least make the font and let us sideload it.

**Comment by olliefr** [comment]
@FANMixco you are not wrong, the official explanation does not make _any_ sense at all! 

Check this out: https://github.com/microsoft/fluentui-emoji/issues/105#issuecomment-1793381202

I think Microsoft should stop ignoring the needs of the many to please the whims of the few. Apple and Google products have country flags. People use them all the time, all over the world!

**Comment by Majestic7979** [comment]
Microsoft, go fuck yourself. Being polite doesn't work. Other manufacturer can display ALL emojis and not turn this political. Screw you for caving in to politicians.

**Comment by medoro** [comment]
Woke and politically correct bullsh*t

**Comment by rudolphos** [comment]
It doesn't make any sense that there are no country flag emojis on Windows ... I can use emojis on Android and iOS, but why not Microsoft Windows ?.

**Comment by Majestic7979** [comment]
I found a solution that worked for myself. I bought a M4 MacBook Pro. Fast, efficient, beautiful and no crap Windows garbage that lacks basic features. Everything I need works on macOS. Microsoft does not care if their consumers are inconvenienced or angered.

**Comment by olliefr** [comment]
@TaylorTWBrown if you are a US resident you can email the [White House](https://www.whitehouse.gov/contact/) instead! 😉 

How can they make America great again if there is no US flag in the world's most popular operating system? 😈

**Comment by ShlomoV5** [comment]
The reason is not political.
Microsoft thinks it has an obligation to educate the public.
Obviously this is because of the progressive "no-identity" idea as if the entire world is one Western, peaceful community with no agenda or disputes. So flags aren't necessary.
I think this is a reason to stop using Windows completly

**Comment by tuannvbg** [comment]
🧵 Ever seen 🇻🇳🇺🇸🇯🇵 show up as plain text on Windows OS?  
This repo fixes that 👇

🔧 `unicode-flags-for-windows` enables full emoji flag display on Windows 10 & 11  
✅ No waiting for updates  
#Unicode #Emoji #Windows #Emojipedia  
https://github.com/tuannvbg/unicode-flags-for-windows

**Comment by marcusleonas** [comment]
bumping this

**Comment by toupswork** [comment]
Microsoft refusing to render country flag emoji is one of those decisions that looks principled only if you squint hard enough to blur the cowardice into architecture.

The technical story is clean: Unicode gives you regional indicator sequences, vendors choose how to render them, Windows chooses not to render them as flags. Fine. Nobody is claiming Satya Nadella personally violated the Unicode Consortium.

But that is exactly what makes it ridiculous.

This is not a hard engineering problem. Windows can draw a taco, a clown face, a pregnant man, five hundred hand gestures, religious symbols, skin tones, family combinations, and every corporate-approved little blob of sanitized expressiveness the emoji-industrial complex can dream up. But the moment two regional indicators might resolve into 🇺🇸 or 🇯🇵 or 🇧🇷, suddenly the world’s most valuable software company becomes a trembling State Department intern.

The absurdity is not that Microsoft made a choice. The absurdity is that Microsoft wants the choice to look like non-choice.

They are not saying, "We cannot render this." They are saying, "We would prefer not to become diplomatically interesting." So instead of a flag, you get a two-letter code, like Windows is wearing a bicycle helmet to cross a carpeted room.

And yes, flags are politically loaded. Obviously. That is what flags are. A flag is politics compressed into a rectangle. It is sovereignty, blood, myth, territory, memory, propaganda, grief, pride, delusion, and bureaucracy all stitched together. That is the entire point. Pretending that this unique danger only appears when the glyph is colorful is insane. The string US is not metaphysically neutral just because it looks uglier.

This is corporate risk management cosplaying as restraint.

A normal user sends a flag because they want a flag. Microsoft intercepts that intent and replaces it with the platform equivalent of "per our geopolitical content policy, please accept these two Latin characters." It is not neutrality. It is symbolic air-gapping. Render the paperwork, not the thing.

And the funniest part is that this does not even avoid the politics. It just produces a different political statement: that every national symbol is too radioactive for the default Windows emoji stack. That is not apolitical. That is a worldview with a compliance department.

The end result is classic enterprise-brain design: technically defensible, legally cautious, user-hostile, aesthetically broken, and somehow proud of itself for confusing risk avoidance with wisdom.

Windows does not omit flag emoji because the system cannot represent them.

It omits them because, somewhere inside the machine, a meeting happened.
